web/client/components/styleeditor/Fields.jsx

```diff
@@ -0,0 +1,472 @@
+                            }
+                            : {
+                                contrastEnhancement: {},
+                                channelSelection: {
+                                    grayChannel: {
+                                        contrastEnhancement: {},
+                                        ...channelSelection?.grayChannel,
+                                        sourceChannelName: newValue
+                                    }
+                                }
+                            });
+                    }
+                    if (key === 'enhancementType') {
+                        return onChange(channelSelectionType === 'auto'
+                            ? {
+                                channelSelection: undefined,
+                                contrastEnhancement: {
+                                    ...value.contrastEnhancement,
+                                    enhancementType: newValue
+                                }
+                            }
+                            : {
+                                contrastEnhancement: {},
+                                channelSelection: Object.keys(channelSelection)
+                                    .reduce((acc, channelKey) => {
+                                        return {
+                                            ...acc,
+                                            [channelKey]: {
+                                                ...channelSelection[channelKey],
+                                                contrastEnhancement: {
+                                                    ...channelSelection[channelKey].contrastEnhancement,
+                                                    enhancementType: newValue
+                                                }
+                                            }
+                                        };
+                                    }, {})
+                            });
+                    }
+                    return null;
+                }}
+            />
+        );
+    },
+    dash: ({
+        label,
+        value,
+        onChange,
+        config: {
+            options
+        }
+    }) => {
+        return (
+            <PropertyField
+                label={label}>
+                <DashArray
+                    dashArray={value}
+                    onChange={onChange}
+                    options={options}
+                    defaultStrokeWidth={2}
+                    isValidNewOption={(option) => {
+                        if (option.label) {
+                            return !option.label.split(' ')
+                                .find((entry) => isNaN(parseFloat(entry)));
+                        }
+                        return false;
+                    }}
+                    creatable
+                />
+            </PropertyField>
+        );
+    }
+};
+
+
+function Fields({
```

Comment:
I see this is the core component of the editor. Can you document usage of the main props ? 
- [ ] properties
- [ ] params
- [ ] config
- [ ] onChange